resources/js/curl-to-go.js

```diff
@@ -17,26 +17,158 @@ function curlToGo(curl) {
+		// curl technically accepted these non-sensical options, they were removed in
+		// https://github.com/curl/curl/commit/913c3c8f
+		'no-http1.0', 'no-http1.1', 'no-http2', 'no-http2-prior-knowledge',
+		'no-tlsv1', 'no-tlsv1.0', 'no-tlsv1.1', 'no-tlsv1.2', 'no-tlsv1.3', 'no-sslv2', 'no-sslv3',
+		'no-ipv4', 'no-ipv6', 'no-proxy-tlsv1', 'no-get', 'no-remote-name', 'no-next',
+		// removed in https://github.com/curl/curl/commit/720ea577
+		'proxy-sslv2', 'no-proxy-sslv2', 'proxy-sslv3', 'no-proxy-sslv3',
+		// removed in https://github.com/curl/curl/commit/388c6b5e
+		// I don't think this was ever a real short option
+		// '~',
+		// renamed to --http2 in https://github.com/curl/curl/commit/0952c9ab
+		'http2.0', 'no-http2.0',
+		// removed in https://github.com/curl/curl/commit/ebf31389
+		// I don't think this option was ever released, it was renamed the same day
+		// it was introduced
+		// 'ssl-no-empty-fragments', 'no-ssl-no-empty-fragments',
+		// renamed to --ntlm-wb in https://github.com/curl/curl/commit/b4f6319c
+		'ntlm-sso', 'no-ntlm-sso',
+		// all options got "--no-" versions in https://github.com/curl/curl/commit/5abfdc01
+		// renamed to --no-keepalive in https://github.com/curl/curl/commit/f866af91
+		'no-keep-alive',
+		// may've been short for --crlf until https://github.com/curl/curl/commit/16643faa
+		// '9',
+		// removed in https://github.com/curl/curl/commit/07660eea
+		// -@ used to be short for --create-dirs
+		'ftp-ascii', // '@',
+		// removed in https://github.com/curl/curl/commit/c13dbf7b
+		// 'c', 'continue',
+		// removed in https://github.com/curl/curl/commit/a1d6ad26
+		// -t used to be short for --upload
+		// 't', 'upload',
+	]);
+
+	// all of curl's short options have a long form
+	var optionAliases = {
+		'0': 'http1.0',
+		'1': 'tlsv1',
+		'2': 'sslv2',
+		'3': 'sslv3',
+		'4': 'ipv4',
+		'6': 'ipv6',
+		'a': 'append',
+		'A': 'user-agent',
+		'b': 'cookie',
+		'B': 'use-ascii',
+		'c': 'cookie-jar',
+		'C': 'continue-at',
+		'd': 'data',
+		'D': 'dump-header',
+		'e': 'referer',
+		'E': 'cert',
+		'f': 'fail',
+		'F': 'form',
+		'g': 'globoff',
+		'G': 'get',
+		'h': 'help',
+		'H': 'header',
+		'i': 'include',
+		'I': 'head',
+		'j': 'junk-session-cookies',
+		'J': 'remote-header-name',
+		'k': 'insecure',
+		'K': 'config',
+		'l': 'list-only',
+		'L': 'location',
+		'm': 'max-time',
+		'M': 'manual',
+		'n': 'netrc',
+		// N is an alias for --no-buffer, not --buffer
+		'N': 'no-buffer',
+		'o': 'output',
+		'O': 'remote-name',
+		'p': 'proxytunnel',
+		'P': 'ftp-port',
+		'q': 'disable',
+		'Q': 'quote',
+		'r': 'range',
+		'R': 'remote-time',
+		's': 'silent',
+		'S': 'show-error',
+		't': 'telnet-option',
+		'T': 'upload-file',
+		'u': 'user',
+		'U': 'proxy-user',
+		'v': 'verbose',
+		'V': 'version',
+		'w': 'write-out',
+		'x': 'proxy',
+		'X': 'request',
+		'Y': 'speed-limit',
+		'y': 'speed-time',
+		'z': 'time-cond',
+		'Z': 'parallel',
+		'#': 'progress-bar',
```

Comment:
`#` is still here.

I was thinking you might've made `-` a boolean option so that if someone were to use `--` to separate the URL from the rest of the arguments, it would interpret `--` as the `'-'` boolean option instead of parsing `curl -- example.com` as `{'-': 'example.com'}`. But I just tried this and it doesn't work

```go
// Generated by curl-to-Go: https://mholt.github.io/curl-to-go

// curl -- example.com

resp, err := http.Get("")
if err != nil {
	// handle err
}
defer resp.Body.Close()
```

it gets parsed as `{"_":["curl"],"example.com":[""]}`.

Reply:
About `#`, my bad.

But yeah, I do seem to remember `-` being useful to have in there, I think it means to use the value being piped through stdin.

Reply:
Right, `-` usually means stdin. For example you can do `curl --stderr -` to make curl print errors to stdout. 

But does adding `'-'` to `boolOptions` actually make sense in the context of the code? When an option gets looked up in `boolOptions` its had its dashes stripped to get just its name. curl-to-Go's bash parsing doesn't interpret a solitary "-" as the option "`-`". The only way I can think of to get into a situation where you're looking up just `'-'` in `boolOptions` is to do `curl --- example.com` which parses as `{"_":["curl","example.com"],"-":true}` if we add `-` to `boolOptions` or as `{"_":["curl"],"-":["example.com"]}` if we don't. But a triple dash has no meaning, I would say that parsing "---" it as the non boolean option "-" is correct, and curl will complain that `---` is an unknown option.

I tried commands like `curl - example.com` (which should produce an error) and `curl --stderr - example.com` and they work the same, regardless of whether or not we have `-` in `boolOptions`. The first case is parsed as `{"_":["curl","example.com"]}` and in the second case `-` is not acting as an option because its preceded by `--stderr`.

Reply:
Interesting; thanks for looking that up and trying all that out. Maybe it was from an early version that didn't handle it well otherwise. Guess it can be removed then.